src/main/java/ModelsTest/UserTest/UserDaoTest.java

```diff
@@ -0,0 +1,93 @@
+package ModelsTest.UserTest;
+
+import Models.User.DuplicateEmailPresentException;
+import Models.User.User;
+import Models.User.UserDao;
+import Models.User.UserNotFoundException;
+import jdk.nashorn.internal.ir.annotations.Ignore;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+import java.io.PrintStream;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertNotNull;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+
+class UserDaoTest {
+
+    private final ByteArrayOutputStream outputStreamCaptor = new ByteArrayOutputStream();
+
+    @BeforeEach
+    void setUp() {
+        System.setOut(new PrintStream(outputStreamCaptor));
+    }
+
+    @AfterEach
+    void tearDown() throws IOException {
+        outputStreamCaptor.close();
+        // TODO: Revert changes made to database in unit test 'Given_ValidUser_Then_InsertUser'
```

Comment:
Should we finish this TODO (so all the tests are passing) before merging into master?

Reply:
I can submit a PR with only my recreateTables() method - which seems a good fit for the tear down between tests

Reply:
(Or you can sneakily the Tables class from my 'audit-logging' branch)

Reply:
Yeah, I'd like to sort these out really. I shall have a stealthy look at your PR this afternoon.